clients/sail_ui/lib/providers/binary_provider.dart

```diff
@@ -164,10 +164,40 @@ class BinaryProvider extends ChangeNotifier {
     _statusController.add(Map.from(_downloadStates));
   }
 
+  /// Check if a binary can be started based on its dependencies
+  String? canStart(Binary binary) {
+    final mainchainReady = mainchainConnected && !inIBD;
+    return switch (binary) {
+      Enforcer() =>
+        mainchainReady ? null : 'Mainchain must be started and fully synced before starting Enforcer',
+      BitWindow() => 
+        enforcerConnected && mainchainReady ? null : 'Mainchain and Enforcer must be running and fully synced before starting BitWindow',
+      Thunder() => 
+        enforcerConnected && mainchainReady ? null : 'Mainchain and Enforcer must be running and fully synced before starting Thunder',
+      Bitnames() => 
+        enforcerConnected && mainchainReady ? null : 'Mainchain and Enforcer must be running and fully synced before starting Bitnames',
+      _ => null, // No requirements for mainchain
+    };
+  }
+
   // Start a binary, and set starter seeds (if set)
   Future<void> startBinary(BuildContext context, Binary binary, {bool useStarter = false}) async {
     if (!context.mounted) return;
 
+    final startError = canStart(binary);
+    if (startError != null) {
+      log.e('Cannot start ${binary.name}: $startError');
+      throw Exception(startError);
+    }
```

Comment:
this check is overkill imo, because it's also added as a check on the 'launch'-button.

but not a big deal, only remove it if you want